src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -116,37 +184,76 @@ def __call__(
-        stage_location: Optional[str] = None,
-        imports: Optional[List[Union[str, Tuple[str, str]]]] = None,
-        packages: Optional[List[Union[str, ModuleType]]] = None,
-        replace: bool = False,
-        if_not_exists: bool = False,
-        parallel: int = 4,
-        execute_as: typing.Literal["caller", "owner"] = "owner",
-        strict: bool = False,
-        external_access_integrations: Optional[List[str]] = None,
-        secrets: Optional[Dict[str, str]] = None,
-        *,
-        statement_params: Optional[Dict[str, str]] = None,
-        source_code_display: bool = True,
-        skip_upload_on_content_match: bool = False,
-    ) -> StoredProcedure:
-        LocalTestOOBTelemetryService.get_instance().log_not_supported_error(
-            external_feature_name="register sproc from file",
-            internal_feature_name="MockStoredProcedureRegistration.register_from_file",
-            parameters_info={},
-            raise_error=NotImplementedError,
-        )
+        import_path: Optional[str] = None,
+        sproc_name: Optional[str] = None,
+    ) -> str:
+        """
+        Imports a python file or a directory of python module structure or a zip of the former.
+        Returns the name of the Python module to be imported.
+        When sproc_name is not None, the import is added to the sproc associated with the name;
+        Otherwise, it is a session level import and will be added to every future sproc call.
+        """
+        file_name, file_extension = os.path.splitext(os.path.basename(file_path))
+
+        is_on_stage = file_path.startswith("@")
+
+        if is_on_stage:
+            stage_registry = self._session._conn.stage_registry
+            stage_name, stage_prefix = extract_stage_name_and_prefix(file_path)
+            local_path = (
+                stage_registry[stage_name]._working_directory + "/" + stage_prefix
+            )
+        else:
+            local_path = file_path
+
+        is_python_import = file_extension in (
+            ".py",
+            ".zip",
+            "",
+        )  # directory is always considered as python module
+
+        if not is_python_import:
+            absolute_module_path = local_path
+            module_name = ""
+        else:
+            if (
+                import_path and not is_on_stage
+            ):  # import_path is only considered for local python files
+                module_root_dir = local_path[
+                    0 : local_path.rfind(import_path.replace(".", "/"))
+                ]
+            elif file_extension == ".py":
+                module_root_dir = os.path.join(local_path, "..")
+            elif file_extension == ".zip":
+                module_root_dir = local_path
+            else:  # directory
+                module_root_dir = os.path.join(local_path, "..")
+
+            absolute_module_path = os.path.abspath(module_root_dir)
+            module_name = file_name.split(".")[
+                0
```

Comment:
looks like this part of code is exactly the same as the one in UDF: https://github.com/snowflakedb/snowpark-python/pull/1335/files

shall we make it a util function?